{{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/analysis/{{cookiecutter.template_analysis_class.lower()}}.py

```diff
@@ -0,0 +1,110 @@
+"""
+{{ cookiecutter.template_analysis_class }} --- :mod:`{{ cookiecutter.package_name }}.analysis.{{ cookiecutter.template_analysis_class }}`
+===========================================================
+
+This module contains the :class:`{{ cookiecutter.template_analysis_class }}` class.
+
+"""
+from typing import Union, TYPE_CHECKING
+
+from MDAnalysis.analysis.base import AnalysisBase
+import numpy as np
+
+if TYPE_CHECKING:
+    from MDAnalysis.core.universe import Universe, AtomGroup
+
+class {{ cookiecutter.template_analysis_class }}(AnalysisBase):
+    """{{ cookiecutter.template_analysis_class }} class.
+
+    This class is used to perform analysis on a trajectory.
+
+    Parameters
+    ----------
+    universe_or_atomgroup: :class:`~MDAnalysis.core.universe.Universe` or :class:`~MDAnalysis.core.groups.AtomGroup`
```

Comment:
What needs changing:
missing the `select` parameter?

What needs consideration is a future revision:
I wonder if this is the access patter we should be encouraging. I.e. I somewhat prefer that folks do selections before they get to the analysis class rather than pass it through (so if they pass universe I'm expecting they want me to use all the atoms, and if they pass an atomgroup it'll be all the atoms in that atomgroup). That being said the core repo is wildly inconsistent about this anyways.

Reply:
I thought the main repo largely followed the `universe_or_atomgroup` and `select` pattern -- it could be good for, e.g., guaranteeing that you're not getting an UpdatingAtomGroup?